back/scripts/datasets/dataset_aggregator.py

```diff
@@ -0,0 +1,150 @@
+
+            self._treat_file(file_infos)
+
+        self._concatenate_files()
+        with open(self.data_folder / "errors.json", "w") as f:
+            json.dump(self.errors, f)
+        return self
+
+    def _treat_file(self, file: tuple) -> None:
+        """
+        Download and normalize a spécific file.
+        """
+        self._download_file(file)
+        self._normalize_file(file)
+
+    def _download_file(self, file_metadata: tuple):
+        """
+        Save locally the output of the URL.
+        """
+        output_filename = self.dataset_filename(file_metadata, "raw")
+        if output_filename.exists():
+            LOGGER.debug(f"File {output_filename} already exists, skipping")
+            return
+        try:
+            urllib.request.urlretrieve(file_metadata.url, output_filename)
+        except HTTPError as error:
+            LOGGER.warning(f"Failed to download file {file_metadata.url}: {error}")
+            msg = (
+                f"HTTP error {error.code} while expecting {file_metadata.resource_status} code"
+            )
+            self.errors[msg].append(file_metadata.url)
+        except Exception as e:
+            LOGGER.warning(f"Failed to download file {file_metadata.url}: {e}")
+            self.errors[str(e)].append(file_metadata.url)
+
+    def dataset_filename(self, file: tuple, step: str):
+        """
+        Expected path for a given file depending on the step (raw or norm).
+        """
+        return (
+            self.data_folder
+            / f"{file.url_hash}_{step}.{file.format if step == 'raw' else 'parquet'}"
+        )
+
+    def _normalize_file(self, file_metadata: tuple) -> pd.DataFrame:
+        out_filename = self.dataset_filename(file_metadata, "norm")
+        if out_filename.exists():
+            LOGGER.debug(f"File {out_filename} already exists, skipping")
+            return
+
+        raw_filename = self.dataset_filename(file_metadata, "raw")
+        if not raw_filename.exists():
+            LOGGER.debug(f"File {raw_filename} does not exist, skipping")
+            return
+        df = self._read_parse_file(file_metadata, raw_filename)
+        if isinstance(df, pd.DataFrame):
+            df.to_parquet(out_filename)
+
+    def _read_parse_file(self, file_metadata: tuple, raw_filename: Path) -> pd.DataFrame | None:
+        raise NotImplementedError()
+
+    def _files_to_run(self):
+        """
+        Select among the input files the ones for which we do not have yet the normalized file.
+        """
+        current = pd.DataFrame(
+            {"filename": [str(x) for x in self.data_folder.glob("*_norm.parquet")], "exists": 1}
+        )
+        return list(
+            self.files_in_scope.merge(
+                current,
+                how="left",
+                on="filename",
+            )
+            .pipe(lambda df: df[df["exists"].isnull()])
+            .drop(columns="exists")
+            .itertuples()
+        )
+
+    def _add_filenames(self):
```

Comment:
method doesn't add, it's computing a new set
```suggestion
    def _get_normalized_filenames(self):
```